flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -843,24 +839,138 @@ private boolean cleanUpJobGraph(JobID jobId, boolean cleanupHA) {
         return false;
     }
 
+    private void cleanupDirtyJobs() {
```

Comment:
It might be worth moving all the Dispatcher's cleanup logic into its own class, e.g. `DispatcherCleanup`.